test/cluster.go

```diff
@@ -51,41 +53,79 @@ func NewCluster() *repository.Cluster {
 	}
 }
 
-// AssertClusters verifies that the `actual` cluster belongs to the `expected` (and compares all fields)
+// NormalizeCluster a function to normalize one or more field in a given cluster
+type NormalizeCluster func(repository.Cluster) repository.Cluster
+
+// AddTrailingSlashes a normalize function which converts all URL by adding a trailing
+// slash if neede
+var AddTrailingSlashes = func(source repository.Cluster) repository.Cluster {
+	source.URL = httpsupport.AddTrailingSlashToURL(source.URL)
+	source.ConsoleURL = httpsupport.AddTrailingSlashToURL(source.ConsoleURL)
+	source.LoggingURL = httpsupport.AddTrailingSlashToURL(source.LoggingURL)
+	source.MetricsURL = httpsupport.AddTrailingSlashToURL(source.MetricsURL)
+	return source
+}
+
+// Normalize returns a new cluster based on the source option, with normalization functions
+// applied
+func Normalize(source repository.Cluster, changes ...NormalizeCluster) repository.Cluster {
+	result := source
+	for _, normalize := range changes {
+		result = normalize(result)
+	}
+	return result
+}
+
+// AssertClusters verifies that the `actual` cluster belongs to the `expected`,
+// and compares all fields including sensitive details if `expectSensitiveInfo` is `true`
 func AssertClusters(t *testing.T, expected []repository.Cluster, actual *repository.Cluster) {
 	for _, e := range expected {
 		if e.ClusterID == actual.ClusterID {
-			AssertEqualClusters(t, &e, actual)
+			AssertEqualCluster(t, &e, actual)
 			return
 		}
 	}
 	// no match found
 	assert.Fail(t, "cluster with ID '%s' couldn't be found in %v", actual.ClusterID, expected)
 }
 
-func AssertEqualClusters(t *testing.T, expected, actual *repository.Cluster) {
-	AssertEqualClusterDetails(t, expected, actual)
+// AssertEqualClusters verifies that all the `actual` and `expected` clusters are have the same values
+// including sensitive details if `expectSensitiveInfo` is `true`
+func AssertEqualClusters(t *testing.T, expected, actual []repository.Cluster) {
+	require.Len(t, actual, len(expected))
+	for _, a := range actual {
+		e := FilterClusterByURL(a.URL, expected)
+		AssertEqualCluster(t, e, &a)
+	}
+}
+
+// AssertEqualCluster verifies that the `actual` and `expected` clusters are have the same values
+// including sensitive details if `expectSensitiveInfo` is `true`
+func AssertEqualCluster(t *testing.T, expected, actual *repository.Cluster) {
 	assert.Equal(t, expected.ClusterID, actual.ClusterID)
+	AssertEqualClusterDetails(t, expected, actual)
 }
 
+// AssertEqualClusterDetails verifies that the `actual` and `expected` clusters are have the same values
+// including sensitive details if `expectSensitiveInfo` is `true`
```

Comment:
You mentioned `expectSensitiveInfo` param in a few places but it looks like it's not defined/actually used anywhere?

Reply:
well, that's the result of a copy/paste from an upcoming PR that actually introduces this arg, but it's not needed here, so I'll remove it from the comment